src/proc/pointcloud.cpp

```diff
@@ -115,13 +114,19 @@ namespace librealsense
 
     void pointcloud::inspect_other_frame(const rs2::frame& other)
     {
-        if (_other_stream != nullptr && other.get_profile().as<rs2::video_stream_profile>() == *_other_stream.get())
+        if (_stream_filter != _prev_stream_filter)
+            _other_stream = nullptr;
+        _prev_stream_filter = _stream_filter;
+
+        if (_extrinsics.has_value() && _other_stream != nullptr && other.get_profile().as<rs2::video_stream_profile>() == *_other_stream.get())
```

Comment:
Using  "(**!**_extrinsics.has_value()  **||** .." would be more readable, i.e. maintainable